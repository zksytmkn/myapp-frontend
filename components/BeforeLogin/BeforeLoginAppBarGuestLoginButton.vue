<template>
  <v-btn
    outlined
    class="ml-2 font-weight-bold"
    color="orange"
    to="/signup"
  >
    <slot>
      {{ $my.pageTitle('signup') }}
    </slot>
  </v-btn>
</template>
