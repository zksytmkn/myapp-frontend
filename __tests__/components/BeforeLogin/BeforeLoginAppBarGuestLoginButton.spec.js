import { shallowMount } from '@vue/test-utils';
import BeforeLoginAppBarGuestLoginButton from '@/components/BeforeLogin/BeforeLoginAppBarGuestLoginButton.vue';

describe('BeforeLoginAppBarGuestLoginButton.vue', () => {
  let wrapper;

  beforeEach(() => {
    // $my オブジェクトをモック
    const $my = {
      pageTitle: jest.fn().mockReturnValue('Signup')
    };

    // コンポーネントをマウントする
    wrapper = shallowMount(BeforeLoginAppBarGuestLoginButton, {
      mocks: {
        $my
      }
    });
  });

  it('renders correctly', () => {
    // コンポーネントが正しくレンダリングされているか確認
    expect(wrapper.element).toMatchSnapshot();
  });

  it('has the correct default properties', () => {
    // デフォルトのプロパティが正しいか確認
    const button = wrapper.find('v-btn');
    expect(button.classes()).toContain('ml-2');
    expect(button.classes()).toContain('font-weight-bold');
    expect(button.attributes('color')).toBe('orange');
    expect(button.attributes('outlined')).toBe('');
  });

  it('has a slot with page title', () => {
    // slotの中にpageTitleが正しく表示されているか確認
    const slotContent = wrapper.find('v-btn').text();
    expect(slotContent).toContain('Signup');
  });

  it('navigates to the correct page on click', () => {
    // クリックしたときに正しいページに遷移するか確認
    const button = wrapper.find('v-btn');
    expect(button.attributes('to')).toBe('/signup');
  });
});
