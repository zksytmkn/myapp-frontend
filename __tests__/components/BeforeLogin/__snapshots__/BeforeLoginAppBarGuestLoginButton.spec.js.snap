// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeforeLoginAppBarGuestLoginButton.vue renders correctly 1`] = `
<v-btn
  class="ml-2 font-weight-bold"
  color="orange"
  outlined=""
  to="/signup"
>
  
    Signup
  
</v-btn>
`;
